src/target/cortexm.c

```diff
@@ -464,6 +464,8 @@ bool cortexm_probe(adiv5_access_port_s *ap)
 		target->designer_code = ap->designer_code;
 		target->part_id = ap->partno;
 	}
+	DEBUG_TARGET("%s: Designer 0x%3x, part_id 0x%4x, AP partno 0x%4x\n", __func__, target->designer_code,
+		target->part_id, ap->partno);
```

Comment:
The format strings should include a 0 term to get them properly suffixed - eg, `%03x`, `%04x`. The designer code is only useful in the event of part identification failure in the error at the bottom of the function, so we'd drop that as duplicate. `part_id` and `partno` are internal names, which we'd like to avoid using in this message - hence the suggestion of `AP part id: %04x, part id: %04x` in the original review note, as this insulates us from internal terminology changes better.

It's an intriguing idea putting this here 🤔 - when we made the original review suggestion, we were thinking it'd go on line 580 (just inside the `case JEP106_MANUFACTURER_STM:` block) given it's only been ST parts that have done bad with their part codes, but it's probably a fair point that more parts deserve this as a just in case. Given that, please move it down to just above `switch (target->designer_code) {`.

Reply:
That's a fair point, though that is because of using ARM's nomenclature for the TARGETID register, rather than internal nomenclature (if memory serves us correctly) - so in that case, `partno` is acceptable.. `part_id` though is internal nomenclature so should be avoided.